crates/red_knot_test/README.md

```diff
@@ -126,6 +126,32 @@ Intervening empty lines or non-assertion comments are not allowed; an assertion
 assertion per line, immediately following each other, with the line immediately following the last
 assertion as the line of source code on which the matched diagnostics are emitted.
 
+## Diagnostic Snapshotting
+
+In addition to inline assertions, one can also snapshot the full diagnostic
+output of a test. This is done by adding the `snapshot-diagnostic` label to the
+code fence info string: For example:
+
+````markdown
+## Unresolvable module import
+
+```py snapshot-diagnostic
+import zqzqzqzqzqzqzq
+```
+````
+
+This will use `insta` to manage an external file snapshot of all diagnostic
+output generated. When any one code block has snapshotting enabled, then it is
+automatically enabled for all code blocks.
+
+Inline assertions, as described above, may be used in conjunction with diagnostic
+snapshotting. However, if there are no inline assertions and diagnostic snapshotting
+is enabled, then no errors will be reported for unmatched diagnostics. This absolves
+one of the need to label each diagnostic manually when it ought to be covered by
```

Comment:
grammar nit: "one of the need"

```suggestion
one of the needs to label each diagnostic manually when it ought to be covered by
```

Though "absolves one of the needs" also sounds a bit strange to my ears -- I'd either go with "absolves the need" or "partially absolves the need"

Reply:
Yeah "needs" sounds weird to me too. I ended up deleting this, but getting rid of "one of" would be a good change here!